### PR TITLE
Add support for Beta Plugins to PluginDashItem

### PR DESCRIPTION
**Pull request by nateweller**
Currently, the `PluginDashItem` component accepts a single `pluginFile` prop, and checks to see if the provided plugin is installed and active when rendering the dash item's content.

This PR adds support to `PluginDashItem` to accept *multiple* plugin files, and then consider the plugin installed or active if *any one* of the provided plugin files are installed or active.

This is then implemented in the existing use of `PluginDashItem` in `DashBoost` to consider both the release and beta plugins, which have differing paths (`jetpack-boost` and `jetpack-boost-dev` respectively).

This ensures a consistent behaviour from `PluginDashItem` whether the user has the release or beta version of Jetpack Boost installed.

Fixes 1164141197617539-as-1201689448751944

#### Changes proposed in this Pull Request:

* Converts `PluginDashItem`'s `pluginFile` string prop to a `pluginFiles` array of strings, so that multiple plugin files can be validated.
* Converts `PluginDashItem`'s `pluginIsActive` and `pluginIsInstalled` props to `aPluginIsActive` and `aPluginIsInstalled`, which are truthy when any single one of the provided plugin file paths is installed or active.
* Updates implementations and tests of `PluginDashItem` to use the new array format for `pluginFiles`, and adds the beta plugin file path in `DashBoost`.

#### Jetpack product discussion

p8oabR-LV-p2#comment-5745

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

For all references to the Jetpack dashboard in these instructions, refer to this path in your WordPress installation: 
`/wp-admin/admin.php?page=jetpack#/dashboard`
1. Install and activate [Jetpack Boost](https://wordpress.org/plugins/jetpack-boost/) from the WordPress plugin repository.
2. Visit the Jetpack dashboard and validate that the Jetpack Boost dash item is displayed like so:
active

3. Deactivate the Jetpack Boost plugin, but keep it installed.
4. Visit the Jetpack dashboard and validate that the Jetpack Boost dash item is displayed like so:

installed

5. Delete the Jetpack Boost plugin. This can also be simulated by renaming the plugin's directory, i.e. `_jetpack-boost`.
6. Visit the Jetpack dashboard and validate that the Jetpack Boost dash item is displayed like so:

uninstalled

7. Install [Jetpack Beta](https://jetpack.com/download-jetpack-beta/) and activate the Release Candidate version of Jetpack Boost. You can find the controls for this here: `/wp-admin/admin.php?page=jetpack-beta&plugin=jetpack-boost`.
8. **Repeat steps 2-4.**

### CODE REVIEW

File: projects/plugins/jetpack/_inc/client/components/plugin-dash-item/index.jsx [hunk 1]
```diff
@@ -174,15 +174,15 @@ PluginDashItem.propTypes = {
```
Comment (elliottprogrammer) [inline projects/plugins/jetpack/_inc/client/components/plugin-dash-item/index.jsx:174]: There is a failing prop type: The prop type `pluginFile` is is marked as required, but its value is `undefined`.

I think prop type `pluginFile` should be changed to `pluginFiles: PropTypes.array.isRequired,`

**Review by elliottprogrammer — COMMENTED** [review]
Tested and it works as expected and as described. 

Everything looks good to me except for the one minor issue with the `propType` in the comment below. Other than that, everything LGTM!  Nice work! 👍
`

**Comment by rcanepa** [comment]
I haven't tested this PR, but I wanted to say that its description and instructions are very well written. Good job, @nateweller !

File: projects/plugins/jetpack/_inc/client/components/plugin-dash-item/index.jsx (hunk 1 above)
Comment (nateweller) [reply projects/plugins/jetpack/_inc/client/components/plugin-dash-item/index.jsx:174]: Thanks for your review, and catching this @elliottprogrammer! 

I've made a fix via fdc4189. I've opted to use `arrayOf` to keep the specific string type for each plugin file path inside of `pluginFiles`:

```javascript
pluginFiles: PropTypes.arrayOf( PropTypes.string ).isRequired,
```

Comment (elliottprogrammer) [reply projects/plugins/jetpack/_inc/client/components/plugin-dash-item/index.jsx:174]: Nice! Even better!  Everything LGTM!

**Pull request merged**